### ISSUE TITLE
QueryParamCollection.Contains does not return true for query params supplied in baseUrl

### ISSUE THREAD

**Issue by GMillerVarian** [tmenier/Flurl#656]
Using Flurl 3.0.2 / Flurl.Http 3.2.0 the following test passes when checking for presence of query param by name only:

```
        [Fact]
        public void FlurlTest()
        {
            // Scenario 1
            var url1 = new Url("https://test.host").SetQueryParam("apiVersion", "1");
            var contains1 = url1.QueryParams.Contains("apiVersion");
            contains1.Should().BeTrue();

            // Scenario 2
            var url2 = new Url("https://test.host?apiVersion=1");
            var contains2 = url2.QueryParams.Contains("apiVersion");
            contains2.Should().BeTrue();
        }
```

However, when including the query param value, the Contains() method returns false in the case where the query param is supplied as part of the baseUrl:

```
        [Fact]
        public void FlurlTest()
        {
            // Scenario 1
            var url1 = new Url("https://test.host").SetQueryParam("apiVersion", "1");
            var contains1 = url1.QueryParams.Contains("apiVersion", "1");
            contains1.Should().BeTrue(); // value is True, assertion passes

            // Scenario 2
            var url2 = new Url("https://test.host?apiVersion=1");
            var contains2 = url2.QueryParams.Contains("apiVersion", "1");
            contains2.Should().BeTrue(); // value is False, assertion fails
        }
```

In debugging this test, I did not not notice any difference in the either the type or value of the query params between the two scenarios.

**Comment by wrexbe** [comment]
```c#
// Scenario 1
object a1 = "hello";
object b1 = "hello";
// a1 == b1 is true
// a1.Equals(b1) is true

// Scenario 2
object a2 = "hello";
object b2 = "_hello".Substring(1,5);
// a2 == b2 is false
// a2.Equals(b2) is true
```

**Comment by tmenier** [comment]
It looks like this was fixed and released in 3.2.1 and I just forgot to close it, but if you think that's wrong feel free to re-open. Sorry about that!
